lib/fluent/plugin/in_forward.rb

```diff
@@ -82,11 +83,15 @@ def shutdown
       @usock.close
       @thread.join
       @lsock.close
+      @connections.each { |c| c.close }
     end
 
     def listen
       log.info "listening fluent socket on #{@bind}:#{@port}"
-      s = Coolio::TCPServer.new(@bind, @port, Handler, @linger_timeout, log, method(:on_message))
+      s = Coolio::TCPServer.new(@bind, @port, Handler, @linger_timeout, log, method(:on_message)) do |connection|
+        @connections.reject! { |conn| conn.closed? }
```

Comment:
Thank you for your comments.

Surely, I think I should guard it by mutex, but please let me ask you a question.
Can this callback be run simultaneously by threads?
It seems that only one thread(`@thread`) handles waiting connections, communicating with accepted connections and replying heartbeat request. So only one thread can run this callback at a time, cannot it?